python_modules/dagster/dagster/_cli/api.py

```diff
@@ -319,7 +320,8 @@ def verify_step(instance, pipeline_run, retry_state, step_keys_to_execute):
 def execute_step_command(input_json):
     with capture_interrupts():
 
-        args = deserialize_as(input_json, ExecuteStepArgs)
+        decompressed_input_json = zlib.decompress(input_json).decode()
```

Comment:
do we get compat issues here?  maybe we need to gracefully handle when we get uncompressed args?

Reply:
Got away without it last time and haven't heard any issues https://github.com/dagster-io/dagster/pull/8564

It would only come up with version skew between the step worker and run worker, which isn't something we explicitly support (only could happen in the very rare case where someone is overriding a step image, in which case it should be on the same version anyways bc if the user code is different things will be very odd).